ndk-build/src/apk.rs

```diff
@@ -96,7 +120,56 @@ impl<'a> UnalignedApk<'a> {
         let lib_path = Path::new("lib").join(abi).join(path.file_name().unwrap());
         let out = self.config.build_dir.join(&lib_path);
         std::fs::create_dir_all(out.parent().unwrap())?;
-        std::fs::copy(path, out)?;
+
+        match self.config.strip {
+            StripConfig::Default => {
+                std::fs::copy(path, out)?;
+            }
+            StripConfig::Strip | StripConfig::Split => {
+                let obj_copy = self.config.ndk.toolchain_bin("objcopy", target)?;
+
+                timeit!(format!("strip debuginfo from {}", lib_path.display()), {
+                    let mut cmd = Command::new(&obj_copy);
+                    cmd.current_dir(&self.config.build_dir);
+                    cmd.arg("--strip-debug");
+                    cmd.arg(path);
+                    cmd.arg(&lib_path);
+
+                    if !cmd.status()?.success() {
+                        return Err(NdkError::CmdFailed(cmd));
+                    }
+                });
+
+                if self.config.strip == StripConfig::Split {
+                    let dwarf_path = {
+                        let mut dp = lib_path.clone();
+                        dp.set_extension("dwarf");
+                        dp
+                    };
```

Comment:
Doesn't [`.with_extension()`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.with_extension) exist for exactly this purpose?